### PR TITLE
Disk: correct volume error - drive is read only

### LINKED ISSUE #85: xStorage - Cannot perform the requested operation when the drive is read only
When attempting to use 2.9.0.0 I keep getting volumes left in a "raw" volumes with assigned driver letters state even when specifying FSFormat. In the logs I am getting the following errors:

VERBOSE: [localhost]:                            [[xDisk]Edrive_xDISK] Set-TargetResource: Creat
ing partition on disk number '2' with drive letter 'E' using all free space.
VERBOSE: [localhost]:                            [[xDisk]Edrive_xDISK] Set-TargetResource: Forma
tting the volume as 'NTFS'.
Cannot perform the requested operation when the drive is read only
    + CategoryInfo          : NotSpecified: (StorageWMI:) [], CimException
    + FullyQualifiedErrorId : StorageWMI 43006,Format-Volume
    + PSComputerName        : localhost
 
VERBOSE: [ocalhost]: LCM:  [ End    Set      ]  [[xDisk]Edrive_xDISK]  in 2.4240 seconds.
The PowerShell DSC resource '[xDisk]Edrive_xDISK' with SourceInfo 
'C:\Users\myprofile\Desktop\TestDSC\SSRSDSC.ps1::71::9::xDisk' threw one or more non-terminating 
errors while running the Set-TargetResource functionality. These errors are logged to the ETW 
channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : NonTerminatingErrorFromProvider
    + PSComputerName        : localhost

Below is an example of the config I am using when getting this error:

        xDisk Hdrive_xDISK
        {
             DiskNumber = 5
             DriveLetter = "H"
             FSlabel = "TempDB"
             FSFormat = "NTFS"
             AllocationUnitSize = 65536
             DependsOn = "[xWaitforDisk]Disk5_xDISK"
        }

I can say that if I keep going back and deleting the volume and running the dsc config it will eventually get a correct NTFS volume, but takes about 3 to 5 times before it won't be in a broken state with a "raw" volume. I have also looked at the disk and confirm they are not in read only. Assuming something in the dsc resource is temporarily putting the disk into a read only mode, but as soon as the config is complete and bombs out the disk is not in read only mode. 

If I drop back down to my previous version I was using successfully (2.6.0.0), this problem does not exist.

### PR DESCRIPTION
**Pull request by JoshuaJSwain**
#### Task list

- [X ] Added an entry under the Unreleased section of the change log in the CHANGELOG.md.
      Entry should say what was changed, and how that affects users (if applicable).
- [ ] Resource documentation added/updated in README.md in the resource folder.
- [ ] Resource parameter descriptions added/updated in schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/storagedsc/190)

### CODE REVIEW

File: Tests/Unit/MSFTDSC_Disk.Tests.ps1 [hunk 1]
```diff
@@ -1275,8 +1275,8 @@ try
 
                 $endTime = Get-Date
 
-                It 'Should take at least 30s' {
-                    ($endTime - $startTime).TotalSeconds | Should -BeGreaterThan 29
+                It 'Should take at least 3s' {
```
[the remark this answers is not in the document]
Comment (PlagueHO) [reply Tests/Unit/MSFTDSC_Disk.Tests.ps1]: I also think we need another new Context fixture that validates that the 3 second minimum timeout is met.

Something like:

Context 'When online GPT disk with no partitions using Disk Number, partition becomes writable after 2 seconds'
  It 'Should take at least 3s'

**Comment by PlagueHO** [comment]
Sorry @JoshuaJSwain - I couldn't use reviewable because something is going weird due to the reverted PR. I actually noticed a few things that needed fixing in this PR.

File: Tests/Unit/MSFTDSC_Disk.Tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoshuaJSwain) [reply Tests/Unit/MSFTDSC_Disk.Tests.ps1]: I've split the tests into two contexts to correct this

**Comment by JoshuaJSwain** [comment]
HI @PlagueHO any further fixes I can work on for this one?

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,14 @@
   CODE\_OF\_CONDUCT.md file.
 - Explicitly removed extra hidden files from release package
```
Comment (PlagueHO) [inline CHANGELOG.md]: Can you remove this blank line?

File: DSCResources/MSFTDSC_Disk/MSFTDSC_Disk.psm1 [hunk 3]
```diff
@@ -443,19 +443,22 @@ function Set-TargetResource
                 After creating the partition it can take a few seconds for it to become writeable
                 Wait for up to 30 seconds for the parition to become writeable
             #>
-            $timeout = (Get-Date) + (New-Timespan -Second 30)
-            while ($partition.IsReadOnly -and (Get-Date) -lt $timeout)
+            $timeAtStart = Get-Date
+            $minimumTimeToWait = $timeAtStart + (New-Timespan -Second 3)
+            $maximumTimeToWait = $timeAtStart + (New-Timespan -Second 30)
+            while (($partitionstate.IsReadOnly -and (Get-Date) -lt $maximumTimeToWait) `
+            -or ((Get-Date) -lt $minimumTimeToWait))
```
Comment (PlagueHO) [inline DSCResources/MSFTDSC_Disk/MSFTDSC_Disk.psm1]: Can you indent this line?

File: DSCResources/MSFTDSC_Disk/MSFTDSC_Disk.psm1 [hunk 4]
```diff
@@ -443,19 +443,22 @@ function Set-TargetResource
                 After creating the partition it can take a few seconds for it to become writeable
                 Wait for up to 30 seconds for the parition to become writeable
             #>
-            $timeout = (Get-Date) + (New-Timespan -Second 30)
-            while ($partition.IsReadOnly -and (Get-Date) -lt $timeout)
+            $timeAtStart = Get-Date
+            $minimumTimeToWait = $timeAtStart + (New-Timespan -Second 3)
+            $maximumTimeToWait = $timeAtStart + (New-Timespan -Second 30)
+            while (($partitionstate.IsReadOnly -and (Get-Date) -lt $maximumTimeToWait) `
```
Comment (PlagueHO) [inline DSCResources/MSFTDSC_Disk/MSFTDSC_Disk.psm1]: Can you add a blank line before this one?

**Comment by JoshuaJSwain** [comment]
thanks @PlagueHO I've pushed the updates

**Comment by PlagueHO** [comment]
:lgtm:

I'm going to squash merge this.

**Comment by JoshuaJSwain** [comment]
I see this in dev, curious to see what the release tempo is for 4.6.0.0. I also work on pr 198 when your unit test pr 201 is merged

**Comment by PlagueHO** [comment]
The DSC Resource kit is released monthly. Usually the week after the community call. So I'd expect the next release date is in the week starting the 1st of April.

**Pull request merged**